bin/utils.sh

```diff
@@ -1,3 +1,12 @@
 #!/usr/bin/env bash
+set -euo pipefail
+IFS=$'\n\t'
 
-${FLUTTER_STORAGE_BASE_URL:=https://storage.googleapis.com} > /dev/null 2>&1
+OS='macos'
```

Comment:
Would be nicer to have a specific check for `Darwin` as people on for example FreeBSD would get the mac version installed instead of crashing. Just an edge case but it irks me that it could be possible.

Reply:
I did not think of make it more flexible since it only supported `linux` and `macos`

```shell
 local versionListUrl="${FLUTTER_STORAGE_BASE_URL}/flutter_infra_release/releases/releases_macos.json"

  if [ "$(uname -s)" == "Linux" ]; then
    versionListUrl="${FLUTTER_STORAGE_BASE_URL}/flutter_infra_release/releases/releases_linux.json"
  fi
```

Reply:
What do you think of the new proposed way